### DISCUSSION TITLE
Show the exception that cause the Retry

### DISCUSSION THREAD

**Discussion by dometec** [eclipse/microprofile-fault-tolerance#617] [Q&A]
I have a method annotated with @Retry and, since some time is invoked more than one time by RetryPolicy (I see the metrics) I would like to know why. So how can I see the exception caught by @Retry?

Thanks
Domenico

**Comment by Azquelt** [comment]
Currently I don't think there's anything in the spec that provides a way for you to see the exception which caused the retry.

The specific implementation that you're using may provide a way to get this data (e.g. through logs or distributed tracing) beyond what's required by the spec.

Alternatively you could log the exception yourself in your own code, but that would somewhat defeat the point of using Fault Tolerance.
